Fix AIP download authn/authz checks

jraddaoui opened the pull request:
Follow the implementation for SIP download made in #1253 and #1266.
Do not overwrite the download endpoint to set the wirte timeout, create
request download and actual download endpoints following the Goa server
stubs. Update dashboard code to request in two steps using a cookie.

Refs #1265.

jraddaoui commented:
@djjuhasz, this is ready for code review if you have some time before leaving.

djjuhasz commented on dashboard/src/components/SipReviewAlert.vue:
```diff
@@ -13,6 +13,7 @@ const emit = defineEmits<{
   (e: "update:expandCounter", value: number): void;
 }>();
 
+// TODO: Load current AIP in aipStore to make download work.
```
Is download not working with this PR then?

jraddaoui replied:
There is an AIP download link in the review alert from the create and review AIP workflow, it's just there where it won't  work for now.

jraddaoui replied:
I fixed this before merging ;)

djjuhasz reviewed (approved):
LGTM @jraddaoui.  There was just one surprising TODO that suggests AIP download isn't working.

jraddaoui committed:
Fix AIP download authn/authz checks

Follow the implementation for SIP download made in #1253 and #1266.
Do not overwrite the download endpoint to set the wirte timeout, create
request download and actual download endpoints following the Goa server
stubs. Update dashboard code to request in two steps using a cookie.

jraddaoui committed:
Fix get SIP extension activity registration

jraddaoui committed:
Fix ingest workflow updated websocket event
